core/include/core/detector.hpp

```diff
@@ -105,33 +106,25 @@ class detector {
     using transform_container =
         array_type<transform_store, mask_id::e_mask_types>;
 
+    detector() = delete;
+
     /** Allowed costructor
-     * @param name the detector
+     * @param name the detector name
      */
     detector(const std::string &name) : _name(name) {}
 
-    /** Copy constructor makes sure the volumes belong to new detector.
-     *
-     * @param other Detector to be copied
-     */
-    detector(const detector &other) = default;
-    detector() = delete;
-    ~detector() = default;
-
     /** Add a new volume and retrieve a reference to it
      *
-     * @param name of the volume
      * @param bounds of the volume, they are expected to be already attaching
      * @param surfaces_finder_entry of the volume, where to entry the surface
      * finder
      *
      * @return non-const reference of the new volume
      */
-    volume &new_volume(const std::string &name,
-                       const array_type<scalar, 6> &bounds,
+    volume &new_volume(const array_type<scalar, 6> &bounds,
```

Comment:
I agree, for detray we should not have volume names ... this was meant for validation, but the volume index should tell us everything we need,